front/src/Phaser/Game/GameScene.ts

```diff
@@ -986,18 +1003,21 @@ ${escapedMessage}
         this.dirty = true;
     }
 
-    private getIndexForTileType(tileType: string): number | null {
-            for (const tileset of this.mapFile.tilesets) {
-                if (tileset.tiles) {
-                    for (const tilesetTile of tileset.tiles) {
-                        if (tilesetTile.type == tileType) {
-                            return tileset.firstgid + tilesetTile.id
-                        }
+    private getIndexForTileType(tileType: string | number): number | null {
+        if (typeof tileType == "number") {
+            return tileType;
+        }
+        for (const tileset of this.mapFile.tilesets) {
```

Comment:
i tihnk we should use a custom property - then we can just use the already existing tile property cache in gameMap (merged recently into develop)